<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="rheaprag" />

<meta name="date" content="2020-10-12" />

<title>01-MissingnessFiltering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">GWAS_QC</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/rheaprag/GWAS_QC">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">01-MissingnessFiltering</h1>
<h4 class="author">rheaprag</h4>
<h4 class="date">2020-10-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-11-17
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>GWAS_QC/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200918code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200918)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200918code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200918)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomrheapragGWASQCtree0cea7d6636cdd675bb99b61e7a0351c6eac4392atargetblank0cea7d6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/rheaprag/GWAS_QC/tree/0cea7d6636cdd675bb99b61e7a0351c6eac4392a" target="_blank">0cea7d6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomrheapragGWASQCtree0cea7d6636cdd675bb99b61e7a0351c6eac4392atargetblank0cea7d6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/rheaprag/GWAS_QC/tree/0cea7d6636cdd675bb99b61e7a0351c6eac4392a" target="_blank">0cea7d6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/bootstrap.min.css
    Untracked:  data/hapmap3.bed
    Untracked:  data/hapmap3.bim
    Untracked:  data/hapmap3.fam
    Untracked:  data/hapmap3.map
    Untracked:  data/hapmap3.ped
    Untracked:  hapmap3.imiss
    Untracked:  hapmap3.lmiss
    Untracked:  hapmap3_01_2.bed
    Untracked:  hapmap3_01_2.bim
    Untracked:  hapmap3_01_2.fam
    Untracked:  hapmap3_01_3.imiss
    Untracked:  hapmap3_01_3.lmiss
    Untracked:  hapmap3_01_4.hwe
    Untracked:  hapmap3_01_5.bed
    Untracked:  hapmap3_01_5.bim
    Untracked:  hapmap3_01_5.fam
    Untracked:  hapmap3_01_6.hwe
    Untracked:  hapmap3_02_1_African.log
    Untracked:  hapmap3_02_1_African.prune.in
    Untracked:  hapmap3_02_1_African.prune.out
    Untracked:  hapmap3_02_1_ColoradoChinese.prune.in
    Untracked:  hapmap3_02_1_ColoradoChinese.prune.out
    Untracked:  hapmap3_02_1_EastAsia.prune.in
    Untracked:  hapmap3_02_1_EastAsia.prune.out
    Untracked:  hapmap3_02_1_Gujarati.prune.in
    Untracked:  hapmap3_02_1_Gujarati.prune.out
    Untracked:  hapmap3_02_1_Mexican.prune.in
    Untracked:  hapmap3_02_1_Mexican.prune.out
    Untracked:  hapmap3_02_1_Toscans.prune.in
    Untracked:  hapmap3_02_1_Toscans.prune.out
    Untracked:  hapmap3_02_1_Utah.prune.in
    Untracked:  hapmap3_02_1_Utah.prune.out
    Untracked:  hapmap3_02_2_African.genome
    Untracked:  hapmap3_02_2_African.log
    Untracked:  hapmap3_02_2_ColoradoChinese.genome
    Untracked:  hapmap3_02_2_EastAsia.genome
    Untracked:  hapmap3_02_2_Gujarati.genome
    Untracked:  hapmap3_02_2_Mexican.genome
    Untracked:  hapmap3_02_2_Toscans.genome
    Untracked:  hapmap3_02_2_Utah.genome
    Untracked:  hapmap3_allpop.genome
    Untracked:  hapmap3_allpop.log
    Untracked:  hapmap3_allpop.prune.in
    Untracked:  hapmap3_allpop.prune.out
    Untracked:  hapmap_African.bed
    Untracked:  hapmap_African.bim
    Untracked:  hapmap_African.fam
    Untracked:  hapmap_African.hh
    Untracked:  hapmap_African.log
    Untracked:  hapmap_African_populations.txt
    Untracked:  hapmap_ChineseColorado_populations.txt
    Untracked:  hapmap_ColoradoChinese.bed
    Untracked:  hapmap_ColoradoChinese.bim
    Untracked:  hapmap_ColoradoChinese.fam
    Untracked:  hapmap_EastAsia.bed
    Untracked:  hapmap_EastAsia.bim
    Untracked:  hapmap_EastAsia.fam
    Untracked:  hapmap_EastAsia_populations.txt
    Untracked:  hapmap_Gujarati.bed
    Untracked:  hapmap_Gujarati.bim
    Untracked:  hapmap_Gujarati.fam
    Untracked:  hapmap_Gujarati_populations.txt
    Untracked:  hapmap_Mexican.bed
    Untracked:  hapmap_Mexican.bim
    Untracked:  hapmap_Mexican.fam
    Untracked:  hapmap_Mexican_populations.txt
    Untracked:  hapmap_Toscans.bed
    Untracked:  hapmap_Toscans.bim
    Untracked:  hapmap_Toscans.fam
    Untracked:  hapmap_Toscans_populations.txt
    Untracked:  hapmap_Utah.bed
    Untracked:  hapmap_Utah.bim
    Untracked:  hapmap_Utah.fam
    Untracked:  hapmap_Utah_populations.txt
    Untracked:  plink
    Untracked:  relationships_w_pops_121708.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/01-MissingnessFiltering.Rmd</code>) and HTML (<code>docs/01-MissingnessFiltering.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/0cea7d6636cdd675bb99b61e7a0351c6eac4392a/analysis/01-MissingnessFiltering.Rmd" target="_blank">0cea7d6</a>
</td>
<td>
rheaprag
</td>
<td>
2020-11-17
</td>
<td>
Update and fix 01
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/1ce1285d06da6cd50ec92dc1c27f133f1c5f98ee/docs/01-MissingnessFiltering.html" target="_blank">1ce1285</a>
</td>
<td>
rheaprag
</td>
<td>
2020-11-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/fc27556b153815c678c584662eccaf8768f01858/analysis/01-MissingnessFiltering.Rmd" target="_blank">fc27556</a>
</td>
<td>
rheaprag
</td>
<td>
2020-11-15
</td>
<td>
Fix formatting in files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/f3159c1be0a84c36530338adb4d21931d0c86c81/docs/01-MissingnessFiltering.html" target="_blank">f3159c1</a>
</td>
<td>
rheaprag
</td>
<td>
2020-11-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/4ef324976db1e2bbe974ad064040236baf64f09f/analysis/01-MissingnessFiltering.Rmd" target="_blank">4ef3249</a>
</td>
<td>
rheaprag
</td>
<td>
2020-11-02
</td>
<td>
Add description to 01
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/de1e8c39036866436e1c02838eb1f5f08186567f/docs/01-MissingnessFiltering.html" target="_blank">de1e8c3</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/75f2559355c650728b0b77f22055f5f90fe5701a/docs/01-MissingnessFiltering.html" target="_blank">75f2559</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/2f21a86fc88a0a034c902fa1a85956c950593144/analysis/01-MissingnessFiltering.Rmd" target="_blank">2f21a86</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
<td>
Fix formating and attempt to adjust plot in 002
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/5e494cadbf2efcd605769e480e730ce7edd2e3e8/docs/01-MissingnessFiltering.html" target="_blank">5e494ca</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/f96e3d88847c8ca57ef387ed1da9608a2e6b0209/analysis/01-MissingnessFiltering.Rmd" target="_blank">f96e3d8</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
<td>
Add formatting to lists
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/13f2bb024c5e17f23298733558e58bdb1085c7fc/docs/01-MissingnessFiltering.html" target="_blank">13f2bb0</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/a849c9a3dafdf7133087bb038b4c439017d35470/analysis/01-MissingnessFiltering.Rmd" target="_blank">a849c9a</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
<td>
Add annotation to file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/046f47d821fade13f3b8cfe39f5dce240bbae05a/docs/01-MissingnessFiltering.html" target="_blank">046f47d</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/f8a7e7d75ed07fb7471741db4250352b150ff0dc/docs/01-MissingnessFiltering.html" target="_blank">f8a7e7d</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/a4ee3034eba90c174e8d06e4c24297b400435cbb/docs/01-MissingnessFiltering.html" target="_blank">a4ee303</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/fbda257a87ddfc0b4f2bdbdf9d70307f27a669cb/analysis/01-MissingnessFiltering.Rmd" target="_blank">fbda257</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-25
</td>
<td>
Add notes to plots in Rmd file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/d83d79d6c44959b24fabc9ad832706ad39be79b1/docs/01-MissingnessFiltering.html" target="_blank">d83d79d</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/7ea97ee01a5da87cdfb61c3c41e6f1411d161eb5/docs/01-MissingnessFiltering.html" target="_blank">7ea97ee</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/40260272d3b0a3bdfe86264aca02902579c5a5cb/analysis/01-MissingnessFiltering.Rmd" target="_blank">4026027</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-25
</td>
<td>
wflow_publish(“analysis/01-MissingnessFiltering.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/965fd727a1cfb2c2cebeb388db696f54f2881496/docs/01-MissingnessFiltering.html" target="_blank">965fd72</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/81708d51d8b4c948fd3656f7a340052d103c342b/analysis/01-MissingnessFiltering.Rmd" target="_blank">81708d5</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Fix formating in 01
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/5f9b28bd09bd2ee975e1c5ce4ca61da963cb7fe5/docs/01-MissingnessFiltering.html" target="_blank">5f9b28b</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/06ca8d7359744dc5ba3f1e325657075092e52c23/docs/01-MissingnessFiltering.html" target="_blank">06ca8d7</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/1b2228ee6cc9a18460c438d13f53d620e1cd7e43/analysis/01-MissingnessFiltering.Rmd" target="_blank">1b2228e</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Fix formating in 01
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/33645b342fa7873f523abfe71ab945c94ed10e9b/docs/01-MissingnessFiltering.html" target="_blank">33645b3</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/f2acc0981f9f44348e9ed6888693872186cb8888/docs/01-MissingnessFiltering.html" target="_blank">f2acc09</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/307ff7ac441f4eb1efb8c5894552d367d4b5b5b9/analysis/01-MissingnessFiltering.Rmd" target="_blank">307ff7a</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Fix formating in 01
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/469ff54b1fe22f1f6c08d4e1e0c83410aa4381be/docs/01-MissingnessFiltering.html" target="_blank">469ff54</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/rheaprag/GWAS_QC/969ca840603d6ca47d9f74d49161a1f06abc7b2a/docs/01-MissingnessFiltering.html" target="_blank">969ca84</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/b6968162c5d788692f2634ab04b86e911a1b2773/analysis/01-MissingnessFiltering.Rmd" target="_blank">b696816</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-18
</td>
<td>
Update 00 and add 01
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In order to remove SNPs that are not genotyped properly, several steps need to be taken.</p>
</div>
<div id="flag-index" class="section level2">
<h2>Flag Index</h2>
<ul>
<li><p><code>--bed</code>: specify the name/path of .bed file</p></li>
<li><p><code>--bfile</code>: specify the name/path of bed/bim/fam files</p></li>
<li><p><code>--bim</code>: specify the name/path of .bim file</p></li>
<li><p><code>--fam</code>: specify the name/path of .fam file</p></li>
<li><p><code>--geno</code>: filter out the variants with a threshold of 0.01 (filter SNPs with a call rate of &lt;99%)</p></li>
<li><p><code>--hardy</code>: writes a list of genotype counts and Hardy-Weinberg equilirbium test statistics to a .hwe file</p></li>
<li><p><code>--hwe</code>: filter out variants with a threshold of 0.000001</p></li>
<li><p><code>--maf</code>: filter out SNPs with a minor allele frequency &lt;0.05</p></li>
<li><p><code>--make-bed</code>: generate bed/bim/fam fileset</p></li>
<li><p><code>--missing</code>: produces sample-based and variant-based missing data reports</p></li>
<li><p><code>--out</code>: specify the name of the output files</p></li>
</ul>
</div>
<div id="determining-the-initial-threshold" class="section level2">
<h2>1. Determining the initial threshold</h2>
<p>The <code>--missing</code> flag will estimate the call rate of individual SNPs within the bfiles (bed/bim/fam) that were generated in the last step.</p>
<pre class="plink"><code>./plink --bfile hapmap3 --missing --out hapmap3</code></pre>
<div id="files-created-in-step-1" class="section level3">
<h3>Files created in Step 1:</h3>
<ul>
<li><p>hapmap3.hh</p></li>
<li><p>hapmap3.imiss</p></li>
<li><p>hapmap3.lmiss</p></li>
<li><p>hapmap3.log</p></li>
</ul>
<p>The important files created are .lmiss and .imiss. These are text files that have the following columns:</p>
<ul>
<li><p><strong>CHR</strong>: Chromosome number</p></li>
<li><p><strong>SNP</strong>: SNP identifier</p></li>
<li><p><strong>N_MISS</strong>: Number of individuals in the data set missing the SNP</p></li>
<li><p><strong>N_GENO</strong>: Number of non-obligatory missing genotypes</p></li>
<li><p><strong>F_MISS</strong>: Proportion of the sample set missing for the SNP</p></li>
</ul>
</div>
<div id="histograms-of-unedited-data-with-initial-threshold" class="section level3">
<h3>Histograms of unedited data with initial threshold:</h3>
<pre><code>lmiss &lt;- read.table(&quot;hapmap3.lmiss&quot;, header = T)
hist(lmiss$F_MISS)</code></pre>
<p><img src="figure/01-MissingnessFiltering.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/13f2bb024c5e17f23298733558e58bdb1085c7fc/docs/figure/01-MissingnessFiltering.Rmd/unnamed-chunk-2-1.png" target="_blank">13f2bb0</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>imiss &lt;- read.table(&quot;hapmap3.imiss&quot;, header = T)
hist(imiss$F_MISS)</code></pre>
<p><img src="figure/01-MissingnessFiltering.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/13f2bb024c5e17f23298733558e58bdb1085c7fc/docs/figure/01-MissingnessFiltering.Rmd/unnamed-chunk-3-1.png" target="_blank">13f2bb0</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="create-new-bfiles-based-on-the-threshold" class="section level2">
<h2>2. Create new bfiles based on the threshold</h2>
<pre class="plink"><code>./plink --bfile hapmap3 --geno 0.01 --maf 0.05 --make-bed --out hapmap3_01_2</code></pre>
<div id="files-created-in-step-2" class="section level3">
<h3>Files created in Step 2:</h3>
<ul>
<li><p>hapmap3_01_2.bed</p></li>
<li><p>hapmap3_01_2.bim</p></li>
<li><p>hapmap3_01_2.fam</p></li>
<li><p>hapmap3_01_2.hh</p></li>
</ul>
</div>
<div id="details" class="section level3">
<h3>Details</h3>
<p>A new threshold is generated with the <code>--geno</code> and <code>--maf</code> plink flags. <code>--geno 0.01</code> filters out SNPs with missing call rates exceeding value of 0.01. <code>maf 0.05</code> filters out SNPs with a minor allele frequency below 0.05. New bflies (bed/bim/fam) with a new threshold are created with the <code>--make-bed</code>.</p>
</div>
</div>
<div id="determine-the-new-threshold-after-filtering" class="section level2">
<h2>3. Determine the new threshold after filtering</h2>
<pre class="plink"><code>./plink --bfile hapmap3_01_2 --geno 0.01 --maf 0.05 --missing --out hapmap3_01_3</code></pre>
<div id="files-created-in-step-3" class="section level3">
<h3>Files created in Step 3:</h3>
<ul>
<li><p>hapmap3_01_3.hh</p></li>
<li><p>hapmap3_01_3.imiss</p></li>
<li><p>hapmap3_01_3.lmiss</p></li>
<li><p>hapmap3_01_3.log</p></li>
</ul>
<p>Similarly to step 1, the .imiss and .lmiss files are created with a new threshold.</p>
</div>
<div id="histograms-with-new-threshold" class="section level3">
<h3>Histograms with new threshold:</h3>
<pre><code>newlmiss &lt;- read.table(&quot;hapmap3_3.lmiss&quot;, header = T)
hist(newlmiss$F_MISS)</code></pre>
<p><img src="figure/01-MissingnessFiltering.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/13f2bb024c5e17f23298733558e58bdb1085c7fc/docs/figure/01-MissingnessFiltering.Rmd/unnamed-chunk-6-1.png" target="_blank">13f2bb0</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-31
</td>
</tr>
<tr>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/7ea97ee01a5da87cdfb61c3c41e6f1411d161eb5/docs/figure/01-MissingnessFiltering.Rmd/unnamed-chunk-6-1.png" target="_blank">7ea97ee</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>newimiss &lt;- read.table(&quot;hapmap3_3.imiss&quot;, header = T)
hist(newimiss$F_MISS)</code></pre>
<p><img src="figure/01-MissingnessFiltering.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/rheaprag/GWAS_QC/blob/7ea97ee01a5da87cdfb61c3c41e6f1411d161eb5/docs/figure/01-MissingnessFiltering.Rmd/unnamed-chunk-7-1.png" target="_blank">7ea97ee</a>
</td>
<td>
rheaprag
</td>
<td>
2020-10-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="plot-generation-and-validation-of-call-rate-distribution" class="section level2">
<h2>4. Plot generation and validation of call rate distribution</h2>
<pre class="plink"><code>./plink --bfile hapmap3_01_2 --hardy --out hapmap3_01_4</code></pre>
<div id="files-created-in-step-4" class="section level3">
<h3>Files created in Step 4:</h3>
<ul>
<li><p>hapmap3_01_4.hh</p></li>
<li><p>hapmap3_01_4.hwe</p></li>
<li><p>hapmap3_01_4.log</p></li>
</ul>
<p>The .hwe file is a text file with the following columns:</p>
<ul>
<li><p><strong>CHR</strong>: Chromosome number</p></li>
<li><p><strong>SNP</strong>: SNP identifier</p></li>
<li><p><strong>TEST</strong>: Indication of what type of test being done, in this case ‘ALL’</p></li>
<li><p><strong>A1</strong>: Allele 1 (minor allele)</p></li>
<li><p><strong>A2</strong>: Allele 2 (major allele)</p></li>
<li><p><strong>GENO</strong>: Separation of genotype counts (AT hom/het/A2 hom)</p></li>
<li><p><strong>O(HET)</strong>: Observed heterozygote frequency</p></li>
</ul>
<p>Plotting the distribution:</p>
<pre><code>hwe &lt;- read.table(&quot;hapmap3_01_4.hwe&quot;, header = T)
hist(hwe$P)</code></pre>
<p><img src="figure/01-MissingnessFiltering.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="filter-the-hardy-weinburg-equilibrium-threshold-and-then-recalculate-statistics" class="section level2">
<h2>5. Filter the Hardy-Weinburg Equilibrium threshold and then recalculate statistics</h2>
<pre class="plink"><code>./plink --bfile hapmap3_01_2 --hwe 0.000001 --make-bed --out hapmap3_01_5</code></pre>
<div id="files-created-in-step-5" class="section level3">
<h3>Files created in Step 5:</h3>
<ul>
<li><p>hapmap3_01_5.bed</p></li>
<li><p>hapmap3_01_5.bim</p></li>
<li><p>hapmap3_01_5.fam</p></li>
<li><p>hapmap3_01_5.hh</p></li>
<li><p>hapmap3_01_5.log</p></li>
</ul>
<p>The <code>hwe 0.000001</code> flag filters out all SNPs with a p-value below 0.000001.</p>
<pre class="plink"><code>./plink --bfile hapmap3_01_5 --hardy --out hapmap3_01_6</code></pre>
<p>This creates the following files:</p>
<ul>
<li><p>hapmap3_01_6.hh</p></li>
<li><p>hapmap3_01_6.hwe</p></li>
<li><p>hapmap3_01_6.log</p></li>
</ul>
<p>Plot the new distribution:</p>
<pre><code>newhwe &lt;- read.table(&quot;hapmap3_01_6.hwe&quot;, header = T)
hist(newhwe$P)</code></pre>
<p><img src="figure/01-MissingnessFiltering.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5      rstudioapi_0.11 whisker_0.4     knitr_1.29     
 [5] magrittr_1.5    R6_2.4.1        rlang_0.4.7     stringr_1.4.0  
 [9] tools_4.0.2     xfun_0.16       git2r_0.27.1    htmltools_0.5.0
[13] ellipsis_0.3.1  rprojroot_1.3-2 yaml_2.2.1      digest_0.6.25  
[17] tibble_3.0.4    lifecycle_0.2.0 crayon_1.3.4    later_1.1.0.1  
[21] vctrs_0.3.4     promises_1.1.1  fs_1.5.0        glue_1.4.1     
[25] evaluate_0.14   rmarkdown_2.3   stringi_1.4.6   compiler_4.0.2 
[29] pillar_1.4.6    backports_1.1.8 httpuv_1.5.4    pkgconfig_2.0.3</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
