---
title: "01-MissingnessFiltering"
author: "rheaprag"
date: "2020-10-12"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In order to remove SNPs that are not genotyped properly, several substeps need to be taken. 

## Flag Index
* `--bed`: specify the name/path of .bed file

* `--bfile`: specify the name/path of bed/bim/fam files

* `--bim`: specify the name/path of .bim file

* `--fam`: specify the name/path of .fam file

* `--geno`: filter out the variants with a threshold of 0.01 (filter SNPs with a call rate of <99%)

* `--hardy`: writes a list of genotype counts and Hardy-Weinberg equilirbium test statistics to a .hwe file 

* `--hwe`: filter out variants with a threshold of 0.001

* `--maf`: filter out SNPs with a minor allele frequency <0.05

* `--make-bed`: generate bed/bim/fam fileset

* `--missing`: produces sample-based and variant-based missing data reports

* `--out`: specify the name of the output files


## 1. Determining the initial threshold
```{plink}
./plink --bed hapmap3_out.bed --bim hapmap3_out.bim --fam hapmap3_out.fam --missing --out hapmap3
```
This creates the following files:

* hapmap3.hh

* hapmap3.imiss

* hapmap3.lmiss

* hapmap3.log  
  
The original data that has not been edited in any way with the initial threshold:
```{}
lmiss <- read.table("hapmap3.lmiss", header = T)
hist(lmiss$F_MISS)
```

```{r, echo = FALSE}
lmiss <- read.table("hapmap3.lmiss", header = T)
hist(lmiss$F_MISS)
```

```{}
imiss <- read.table("hapmap3.imiss", header = T)
hist(imiss$F_MISS)
```

```{r, echo = FALSE}
imiss <- read.table("hapmap3.imiss", header = T)
hist(imiss$F_MISS)
```

## 2. Create new bfiles based on the threshold
```{plink}
./plink --bed hapmap3_out.bed --bim hapmap3_out.bim --fam hapmap3_out.fam --geno 0.01 --maf 0.05 --make-bed --out hapmap3_2
```
This creates the following files:

* hapmap3_2.bed

* hapmap3_2.bim

* hapmap3_2.fam

* hapmap3_2.hh

## 3. Determine the new threshnold after filtering
```{plink}
./plink --bfile hapmap3_2 --geno 0.01 --maf 0.05 --missing --out hapmap3_3
```
This creates the following files:

* hapmap3_3.hh

* hapmap3_3.imiss

* hapmap3_3.lmiss

* hapmap3_3.log

## 4. Plot generation and validation of call rate distribution
```{plink}
./plink --bfile hapmap3_2 --hardy --out hapmap3_4
```
This creates the following files:

* hapmap3_4.hh

* hapmap3_4.hwe

* hapmap3_4.log


The data at this point after filtering with the new threshold:
```{}
newlmiss <- read.table("hapmap3_3.lmiss", header = T)
hist(newlmiss$F_MISS)
```

```{r, echo = FALSE}
newlmiss <- read.table("hapmap3_3.lmiss", header = T)
hist(newlmiss$F_MISS)
```

```{}
newimiss <- read.table("hapmap3_3.imiss", header = T)
hist(newimiss$F_MISS)
```

```{r, echo = FALSE}
newimiss <- read.table("hapmap3_3.imiss", header = T)
hist(newimiss$F_MISS)
```

## 5. Filter the Hardy-Weinburg Equilibrium threshold and then recalculate statistics
```{plink}
./plink --bfile hapmap3_2 --hwe 0.0001 --make-bed --out hapmap3_5
```
This creates the following files:

* hapmap3_5.bed

* hapmap3_5.bim

* hapmap3_5.fam

* hapmap3_5.hh

* hapmap3_5.log

```{plink}
./plink --bfile hapmap3_5 --hardy --out hapmap3_final
```
This creates the following files:

* hapmap3_final.hh
  
* hapmap3_final.hwe
  
* hapmap3_final.log

