---
title: "01-MissingnessFiltering"
author: "rheaprag"
date: "2020-10-12"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction

In order to remove SNPs that are not genotyped properly, several steps need to be taken. 

## Flag Index
* `--bed`: specify the name/path of .bed file

* `--bfile`: specify the name/path of bed/bim/fam files

* `--bim`: specify the name/path of .bim file

* `--fam`: specify the name/path of .fam file

* `--geno`: filter out the variants with a threshold of 0.01 (filter SNPs with a call rate of <99%)

* `--hardy`: writes a list of genotype counts and Hardy-Weinberg equilirbium test statistics to a .hwe file 

* `--hwe`: filter out variants with a threshold of 0.000001

* `--maf`: filter out SNPs with a minor allele frequency <0.05

* `--make-bed`: generate bed/bim/fam fileset

* `--missing`: produces sample-based and variant-based missing data reports

* `--out`: specify the name of the output files


## 1. Determining the initial threshold
The `--missing` flag will estimate the call rate of individual SNPs within the bfiles (bed/bim/fam) that were generated in the last step. 
```{plink}
./plink --bfile hapmap3 --missing --out hapmap3
```

### Files created in Step 1:

* hapmap3.hh

* hapmap3.imiss

* hapmap3.lmiss

* hapmap3.log  

The important files created are .lmiss and .imiss. These are text files that have the following columns:

* **CHR**: Chromosome number

* **SNP**: SNP identifier

* **N_MISS**: Number of individuals in the data set missing the SNP

* **N_GENO**: Number of non-obligatory missing genotypes

* **F_MISS**: Proportion of the sample set missing for the SNP
  
### Histograms of unedited data with initial threshold:
```{}
lmiss <- read.table("hapmap3.lmiss", header = T)
hist(lmiss$F_MISS)
```

```{r, echo = FALSE}
lmiss <- read.table("hapmap3.lmiss", header = T)
hist(lmiss$F_MISS)
```

```{}
imiss <- read.table("hapmap3.imiss", header = T)
hist(imiss$F_MISS)
```

```{r, echo = FALSE}
imiss <- read.table("hapmap3.imiss", header = T)
hist(imiss$F_MISS)
```

## 2. Create new bfiles based on the threshold
```{plink}
./plink --bfile hapmap3 --geno 0.01 --maf 0.05 --make-bed --out hapmap3_01_2
```
### Files created in Step 2:

* hapmap3_01_2.bed

* hapmap3_01_2.bim

* hapmap3_01_2.fam

* hapmap3_01_2.hh

### Details
A new threshold is generated with the `--geno` and `--maf` plink flags. `--geno 0.01` filters out SNPs with missing call rates exceeding value of 0.01. `maf 0.05` filters out SNPs with a minor allele frequency below 0.05. New bflies (bed/bim/fam) with a new threshold are created with the `--make-bed`. 

## 3. Determine the new threshold after filtering
```{plink}
./plink --bfile hapmap3_01_2 --geno 0.01 --maf 0.05 --missing --out hapmap3_01_3
```
### Files created in Step 3:

* hapmap3_01_3.hh

* hapmap3_01_3.imiss

* hapmap3_01_3.lmiss

* hapmap3_01_3.log

Similarly to step 1, the .imiss and .lmiss files are created with a new threshold.

## 4. Plot generation and validation of call rate distribution
```{plink}
./plink --bfile hapmap3_01_2 --hardy --out hapmap3_01_4
```
### Files created in Step 4:

* hapmap3_01_4.hh

* hapmap3_01_4.hwe

* hapmap3_01_4.log


The .hwe file is a text file with the following columns:

* **CHR**: Chromosome number

* **SNP**: SNP identifier

* **TEST**: Indication of what type of test being done, in this case 'ALL'

* **A1**: Allele 1 (minor allele)

* **A2**: Allele 2 (major allele)

* **GENO**: Separation of genotype counts (AT hom/het/A2 hom)

* **O(HET)**: Observed heterozygote frequency


### Histograms with new threshold:
```{}
newlmiss <- read.table("hapmap3_3.lmiss", header = T)
hist(newlmiss$F_MISS)
```

```{r, echo = FALSE}
newlmiss <- read.table("hapmap3_01_3.lmiss", header = T)
hist(newlmiss$F_MISS)
```

```{}
newimiss <- read.table("hapmap3_3.imiss", header = T)
hist(newimiss$F_MISS)
```

```{r, echo = FALSE}
newimiss <- read.table("hapmap3_01_3.imiss", header = T)
hist(newimiss$F_MISS)
```

## 5. Filter the Hardy-Weinburg Equilibrium threshold and then recalculate statistics
```{plink}
./plink --bfile hapmap3_01_2 --hwe 0.000001 --make-bed --out hapmap3_01_5
```
### Files created in Step 5:

* hapmap3_01_5.bed

* hapmap3_01_5.bim

* hapmap3_01_5.fam

* hapmap3_01_5.hh

* hapmap3_01_5.log

The `hwe 0.000001` flag filters out all SNPs with a p-value below 0.000001.

```{plink}
./plink --bfile hapmap3_01_5 --hardy --out hapmap3_01_6
```
This creates the following files:

* hapmap3_01_6.hh
  
* hapmap3_01_6.hwe
  
* hapmap3_01_6.log

```{r, echo= FALSE}
hwe <- read.table("hapmap3_01_6.hwe", header = T)
hist(hwe$P)
```
