---
title: "02-RelatednessFiltering"
author: "rheaprag"
date: "2020-10-18"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Introduction
Relatedness filtering removes the duplicate records and individuals that are highly related. 

## Flag Index
* `--autosome`: excludes all unplaced and non-autosomal variants

* `--bed`: specify the name/path of .bed file

* `--bfile`: specify the name/path of bed/bim/fam files

* `--bim`: specify the name/path of .bim file

* `--extract`: filter out variants that are not in the file provided

* `--fam`: specify the name/path of .fam file

* `--geno`: filter out the variants with a threshold of 0.01 (filter SNPs with a call rate of <99%)

* `--genome`: create IBD report

* `--het`: compute the observed and expected autosomal homozygous genotype values for each sample and writes it in .het file

* `--make-bed`: generate bed/bim/fam file set

* `--out`: specify the name of the output files

## 1. Create a pruned list of SNP IDs
Once the populations are separated, the data can go onto the next filtering step.

```{}
./plink --bfile hapmap_African --autosome --indep-pairwise 50 5 0.2 --out hapmap3_02_1_African
```
### Files created in Step 1:
  
* hapmap3_02_1_African.log

* hapmap3_02_1_African.prune.in
  
* hammap3_02_1_African.prune.out

## 2. Determine the IBD of samples
```{}
./plink --bfile hapmap_African --extract hapmap3_02_1_African.prune.in --genome --out hapmap3_02_2_African
```
### Files created in Step 2:

* hapmap3_02_2_African.genome
  
* hapmap3_02_2_African.hh
  
* hapmap3_02_2_African.log
  
In order to check the IBD:
```{}
library("ggplot2")
ibd <- read.table("hapmap3_02_2_African.genome", header = T)
ggplot(data=ibd,aes(x=Z0,y=Z1,color=RT))+geom_point(alpha=1/4)+theme_bw()
```

```{r, echo = FALSE}
library("ggplot2")
ibd <- read.table("hapmap3_02_2_African.genome", header = T)
ggplot(data=ibd,aes(x=Z0,y=Z1,color=RT))+geom_point(alpha=1/4)+theme_bw()
```

The rest of the IBD plots for the population subsets can be seen [here](IBD_plots.html).

## 3. Check the heterozygosity in the samples
```{}
./plink --bed hapmap3_5.bed --bim hapmap3_5.bim --fam hapmap3_5.fam --het --out hapmap3_02_3
```
### Files created in Step 3:

* hapmap3_02_3.het
  
* hapmap3_02_3.hh
  
* hapmap3_02_3.log

## 4. LD prune SNPs from bfiles
```{}
./plink --bed hapmap3_5.bed --bim hapmap3_5.bim --fam hapmap3_5.fam --extract hapmap3.prune.in --make-bed --out hapmap3_02_4
```
### Files created in Step 4:

* hapmap3_02_4.bed
  
* hapmap3_02_4.bim
  
* hapmap3_02_4.fam
  
* hapmap3_02_4.hh
  
* hapmap3_02_4.log

## 5. Remove dumplicate or related samples & regenerate heterozygosity estimates from the filtered samples
```{}
./plink --bfile hapmap3_02_4 -het --out hapmap3_02_5
```
### Files created in Step 5:

* hapmap3_02_5.het
  
* hapmap3_02_5.hh
  
* hapmap3_02_5.log





